EIPS/eip-2929.md

```diff
@@ -0,0 +1,165 @@
+
+When an address is either the target of a (`EXTCODESIZE`, `EXTCODECOPY`, `EXTCODEHASH` or `BALANCE`) opcode or the target of a (`CALL`, `CALLCODE`, `STATICCALL`, `DELEGATECALL`) opcode, the gas costs are computed as follows:
+
+* If the target is not in `accessed_addresses`, charge `COLD_ACCOUNT_ACCESS_COST` gas, and add the address to `accessed_addresses`.
+* Otherwise, charge `WARM_STORAGE_READ_COST` gas.
+
+In all cases, the gas cost is charged and the map is updated at the time that the opcode is being called. When a `CREATE` or `CREATE2` opcode is called, immediately (ie. before checks are done to determine whether or not the address is unclaimed) add the address being created to `accessed_addresses`, but gas costs of `CREATE` and `CREATE2` are unchanged.
+
+For `SLOAD`, if the `(address, storage_key)` pair (where `address` is the address of the contract whose storage is being read) is not yet in `accessed_storage_keys`, charge `COLD_SLOAD_COST` gas and add the pair to `accessed_storage_keys`. If the pair is already in `accessed_storage_keys`, charge `WARM_STORAGE_READ_COST` gas.
+
+### SSTORE changes
+
+When calling `SSTORE`, check if the `(address, storage_key)` pair is in `accessed_storage_keys`. If it is not, charge an additional `COLD_SLOAD_COST` gas, and add the pair to `accessed_storage_keys`. Additionally, modify the parameters defined in [EIP 2200](https://eips.ethereum.org/EIPS/eip-2200) as follows:
+
+| Parameter | Old value | New value |
+| - | - | - |
+| `SLOAD_GAS` | 800 | `= WARM_STORAGE_READ_COST` |
+| `SSTORE_RESET_GAS` | 5000 | `5000 - COLD_SLOAD_COST` |
+
+The other parameters defined in EIP 2200 are unchanged.
+
+### SELFDESTRUCT changes
+
+If the ETH recipient of a `SELFDESTRUCT` is not in `accessed_addresses` (regardless of whether or not the amount sent is nonzero), charge an additional `COLD_ACCOUNT_ACCESS_COST` on top of the existing gas costs, and add the ETH recipient to the set.
+
+### Contract breakage mitigations
+
+See the Security Considerations section for details on the tradeoffs between these versions.
+
+#### Version 1
+
+Do nothing (ie. implement only the changes described above).
+
+#### Version 2
+
+Add a `POKE` precompile, with cost 4500 gas, that takes two stack arguments, `address` and `storage_slot`, and adds `address` to `accessed_addresses` and `(address, storage_slot)` to `accessed_storage_keys`.
+
+#### Version 3
+
+Access lists: see [EIP 2930](https://github.com/ethereum/EIPs/pull/2930)
+
```

Comment:
```suggestion
```

EIPs should be assertions, not discussions.  We have git history for tracking history of changes and we have the discussion-to link for discussing proposed future changes to the current specification.  Consider writing the EIP with a reader 5 years from now in mind, where they are just trying to implement or understand a specification, and they don't really care at all about the iteration that happened prior to the EIP being final.